libraries/AP_EFI/AP_EFI_Serial_FH.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_EFI_config.h"
+
+#if AP_EFI_SERIAL_FH_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/utility/sparse-endian.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+
+#include "AP_EFI_Serial_FH.h"
+
+#include <stdio.h>
+
+// RPM Threshold for fuel consumption estimator
+#define RPM_THRESHOLD                100
+
+extern const AP_HAL::HAL &hal;
+
+
+AP_EFI_Serial_FH::AP_EFI_Serial_FH(AP_EFI &_frontend):
+    AP_EFI_Backend(_frontend)
+{
+    port = AP::serialmanager().find_serial(AP_SerialManager::SerialProtocol_EFI, 0);
+
+    // Indicate that temperature and fuel pressure are supported
+    internal_state.fuel_pressure_status = Fuel_Pressure_Status::OK;
+    internal_state.temperature_status = Temperature_Status::OK;
+
+    // FlyHenry ECU reports EGT 
+    internal_state.cylinder_status.exhaust_gas_temperature = 0;
+    internal_state.cylinder_status.exhaust_gas_temperature2 = 0;
+
+}
+
+
+  bool AP_EFI_Serial_FH::process(uint32_t const now) {
+    if (port != nullptr) {
+      if ((ind && (now - last_recv > 300)) || (ind >= sizeof(pkt))) ind = 0;
+      size_t n = MIN(sizeof(pkt) - ind, port->available());
+      if (n) {
+        last_recv = now;
+        while (n--) {
+          uint8_t i = ind;
+          pkt[ind++] = port->read();
+          const uint8_t hdr[] = { 0xA5, 0x5A, 1, 0 };
+          if (i < sizeof(hdr) && pkt[i] != hdr[i]) ind = 0;
+          else if (sizeof(pkt) == ind) return (0xD == data.eop && data.sum == sum8(pkt, 48));
+        }
+      }
+    }
+    return false;
+  }
+
+static unsigned char boza = 0;

```

Comment:
This thing makes no sense to me.  It will be 0x66 some of the time even when we haven't processed a packet.  Since the call rate of update() isn't specified anywhere, that's essentially at random.